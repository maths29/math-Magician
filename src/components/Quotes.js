import { useEffect, useState } from 'react';
import '../styles/quotes.css';
import Header from './Header';

const Quotes = () => {
  const [quoteData, setQuote] = useState({});
  const [error, setError] = useState('');
  const header = new Headers();
  header.append('x-api-key', 'OVLcX2/3XpWrpr1fR1+FqA==HqgZvSKqav5X8B74');
  const URL = 'https://api.api-ninjas.com/v1/quotes?category=';
  const handlefetchApi = () => {
    const controller = new AbortController();
    const { signal } = controller;

    fetch(URL, {
      signal,
      method: 'GET',
      headers: header,
    })
      .then((res) => res.json())
      .then((data) => {
        setQuote(data[0]);
      })
      .catch((e) => {
        setError(`error message ${e}`);
      });
    return () => {
      setError('');
      controller.abort();
    };
  };
  useEffect(() => handlefetchApi(), []);

  return (
    <>
      <Header />
      <div>
        <div className="pDiv">
          <p>
            {quoteData?.quote}
          </p>
        </div>
        <h4 style={{ color: 'white', display: 'none' }}>{error}</h4>
      </div>
    </>
  );
};
export default Quotes;
