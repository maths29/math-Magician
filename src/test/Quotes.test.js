import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Quotes from '../components/Quotes';

describe('test the quotes loading', () => {
  test('check the Quotes snapshots', () => {
    const tree = renderer.create(
      <Router>
        <Quotes />
      </Router>,
    ).toJSON;

    expect(tree).toMatchSnapshot();
  });
});
