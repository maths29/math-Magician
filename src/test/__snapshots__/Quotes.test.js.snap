// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test the quotes loading check the Quotes snapshots 1`] = `[Function]`;
